### DISCUSSION TITLE
Model Values in Filenames

### DISCUSSION THREAD

**Discussion by henrylawson** [shrinerb/shrine#638] [Q&A] [answered]
**Question**
Is there an easy way to customize the filename for the derivative images?

I am aware of the "Pretty Location" feature (https://shrinerb.com/docs/plugins/pretty_location).

I would like to go one step further and customize the filename to have a per-attachment value from the model (shown as "customvalue" in the mock below:

```
# :model/:id/:attachment/:customvalue-:derivative-:uid.:extension 
```

**Reason**
I would like to have derivatives generated that have "descriptive" filenames to better satisfy SEO recommendations https://developers.google.com/search/docs/appearance/google-images#descriptive-titles-captions-filenames

**Comment by janko** [comment, marked as answer]
You can override the `generate_location` method - https://shrinerb.com/docs/getting-started#location

**Reply by jrochkind** [reply]
Here's an example from some of my open source code!

https://github.com/sciencehistory/scihist_digicoll/blob/3fc6cb80643c79c80b63f5af2c5d231f79525843/app/uploaders/combined_audio_uploader.rb#L15-L21

**Reply by henrylawson** [reply]
Ah, thank you for pointing me to those docs, that would work perfectly.
